### ISSUE TITLE
[NFR] Skip other validators if field is empty and optional

### ISSUE THREAD

**Issue by Agent-J** [phalcon/cphalcon#690]
If field is empty and 'required' validator is absent then other validators should be skipped.
This is a very common scenario: the field must either be empty or have a valid format.

```php
use Phalcon\Validation\Validator\PresenceOf,
    Phalcon\Validation\Validator\Email;

$validation = new Phalcon\Validation;
$validation->add('email_1', new Email);
$validation->add('email_2', new Email);
$validation->add('email_2', new PresenceOf);

// The validation should be passed if email_1 field is empty 'cause it optional
```

**Comment by crocodile2u** [comment]
Basically, all of the phalcon validation is useless without this simple feature.

**Comment by phalcon** [comment]
@crocodile2u This simple feature is implemented in Phalcon 2

**Comment by crocodile2u** [comment]
Wow, that's really-really nice to hear! Thanks for your work!

**Comment by Green-Cat** [comment]
@andresgutierrez Implemented in 2.0.
